sdk/run.go

```diff
@@ -146,6 +149,32 @@ func invoke(doc *attest.AttestationDoc, conn *websocket.Conn, data []byte) (*cli
 		return nil, err
 	}
 	log.Debugf("< Received Function Results.")
+	resData.DecodedAttestationDocument = attestDoc.decoded
+	resData.RawAttestationDocument = attestDoc.raw
+
+	log.Debugf("* Verifying Function Results.")
+
+	// TODO -- connect is already doing this
+	var ud AttestationUserData
+	if err := json.Unmarshal(attestDoc.decoded.UserData, &ud); err != nil {
+		return nil, err
+	}
+
+	publicKey, err := x509.ParsePKCS1PublicKey(ud.SignatureVerificationKey)
+	if err != nil {
+		return nil, err
+	}
+
+	c := sha256.New()
+	if err := json.NewEncoder(c).Encode(resData.Checksums); err != nil {
+		return nil, err
+	}
+
+	if err := rsa.VerifyPSS(publicKey, crypto.SHA256, c.Sum(nil), resData.SignedResults, nil); err != nil {
```

Comment:
PCRs use Sha384, we could use that as well to be consistent. Not a big deal though